File: backend/src/main/java/com/radar/prospection/domain/Signal.java
package com.radar.prospection.domain;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "signals")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Signal {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Enumerated(EnumType.STRING)
    private SignalType type;

    private String company;
    private String title;

    @Column(length = 3000)
    private String description;

    // Pour les levées de fonds
    private String fundingAmount;
    private String fundingRound;

    // Pour les nominations
    private String personName;
    private String newRole;

    // Score d'opportunité calculé par Claude
    @Column(name = "opportunity_score")
    private Integer opportunityScore;

    @Column(name = "opportunity_reason", length = 1000)
    private String opportunityReason;

    @Column(name = "suggested_action", length = 500)
    private String suggestedAction;

    private String url;
    private String sourceWebsite;

    @Column(name = "detected_at")
    @Builder.Default
    private LocalDateTime detectedAt = LocalDateTime.now();

    @Enumerated(EnumType.STRING)
    @Builder.Default
    private SignalStatus status = SignalStatus.NEW;
}

File: backend/src/main/kotlin/com/radar/prospection/domain/Signal.kt
package com.radar.prospection.domain

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "signals")
class Signal(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @Enumerated(EnumType.STRING)
    var type: SignalType? = null,

    var company: String? = null,
    var title: String? = null,

    @Column(length = 3000)
    var description: String? = null,

    var fundingAmount: String? = null,
    var fundingRound: String? = null,
    var personName: String? = null,
    var newRole: String? = null,

    @Column(name = "opportunity_score")
    var opportunityScore: Int? = null,

    @Column(name = "opportunity_reason", length = 1000)
    var opportunityReason: String? = null,

    @Column(name = "suggested_action", length = 500)
    var suggestedAction: String? = null,

    var url: String? = null,
    var sourceWebsite: String? = null,

    @Column(name = "detected_at")
    var detectedAt: LocalDateTime = LocalDateTime.now(),

    @Enumerated(EnumType.STRING)
    var status: SignalStatus = SignalStatus.NEW
)
